packages/test/local-server-tests/src/test/noDeltaStream.spec.ts

```diff
@@ -117,11 +117,11 @@ describe("No Delta Stream", () => {
         assert.strictEqual(container.readonlyPermissions, true, "container.readonlyPermissions");
         assert.ok(container.readOnlyInfo.readonly, "container.storageOnly");
 
-        const deltaManager = container.deltaManager as DeltaManager;
+        const deltaManager = container.deltaManager;
         assert.strictEqual(deltaManager.active, false, "deltaManager.active");
         assert.strictEqual(deltaManager.readonly, true, "deltaManager.readonly");
-        assert.strictEqual(deltaManager.readonlyPermissions, true, "deltaManager.readonlyPermissions");
-        assert.strictEqual(deltaManager.connectionMode, "read", "deltaManager.connectionMode");
+        // TODO: remove as any cast once IDeltaManager has connectionMode
+        assert.strictEqual((deltaManager as any).connectionMode, "read", "deltaManager.connectionMode");
```

Comment:
This cast is weird, given that you are exposing connectionMode.

Reply:
This has to be a two PR process since I need to deploy the changes to `IDeltaManager` to `container-definitions` and then consume it here in packages.

Reply:
It's probably fine to have it there. We are not fully consistent here. I.e. for 99% of the runtime code, we hide difference between "connected" and "connecting" states, and that's on purpose. But we do expose events and runtime does listen to them (OpPerfTelemetry being the only place though) - see on("connect")).
So it's probably fine to leave it as is, but I like to keep API small(er) as that has smaller support cost over long period of time...

Reply:
So it would be better (in the long run) to simply not have connectionMode exposed in any interface?

Reply:
"any" interface is too wide. It would be great not to expose difference between connected & connecting states to runtime, yes. If test code needs it, it's fine to expose it in some form or shape. Note that test code might have broader access (like access to Container object) which would expose same info, so maybe we can avoid it altogether (it's Ok to expose it on IContainer if it's not exposed there)

Reply:
Leaving the cast should decrease our API surface. The other options considered are adding `connectionMode?:` to either `IDeltaManager` or a new interface.